src/app/navigation/utils.ts

```diff
@@ -109,3 +111,32 @@ export const findPathForParams = (params: object, paths: string[]) => {
       paramsInPath.every(({name}) => paramKeys.includes(name.toString()));
   });
 };
+
+export const isScrollTarget = (
+  object: { [key: string]: any }
+): object is ScrollTarget => {
+  if (
+    !object.elementId
+    || typeof object.elementId !== 'string'
+    || typeof object.type !== 'string'
+  ) { return false; }
+  return true;
+};
+
+export const getScrollTargetFromQuery = (
+  query: OutputParams,
+  hash: string
+): ScrollTarget | null => {
+  if (!hash || !query.target || Array.isArray(query.target)) { return null; }
+  let parsed: any;
+  try {
+    parsed = JSON.parse(decodeURIComponent(query.target));
+  } catch {
+    return null;
+  }
+  if (isPlainObject(parsed)) {
+    (parsed as {[key: string]: any}).elementId = hash.replace('#', '');
```

Comment:
this deserialization logic is sort of related to the serialization logic that exists in the ContentLink component, it would be nice if there were helpers for these that lived near one another